### PR TITLE
Support root= being set in dracut embedded cmdline better^Udracut and manual cleanups

### PR DESCRIPTION
**Pull request by nabijaczleweli**
### Motivation and Context
#13088

### Description
See individual commits. ~~All but the top~~ NFC.

### How Has This Been Tested?
~~Not. I don't currently have a system I could test this on.~~ See above.

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:
- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [x] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes. – none apply
- [ ] I have run the ZFS Test Suite with this change applied. – CI take my hand
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by nabijaczleweli** [comment]
Rebased

File: contrib/dracut/90zfs/parse-zfs.sh.in [hunk 1]
```diff
@@ -16,6 +16,7 @@ else
 fi
 
 wait_for_zfs=0
+root="$(getarg root=)"
```
[the remark this answers is not in the document]
Comment (nabijaczleweli) [reply contrib/dracut/90zfs/parse-zfs.sh.in:19]: Hm, yeah. It seems I overfit to the original "solution", then gave up after failing to reproduce – your analysis appears to be correct, and lines up with what I managed to dig up later, esp. given the single space, which I appear to've missed. I'll drop this patch.

[the remark this answers is not in the document]
Comment (zdykstra) [reply contrib/dracut/90zfs/parse-zfs.sh.in]: That's a really good question. I've never seen that behavior before. I'll see if I can setup a test environment locally to try to reproduce the problem with an Arch VM.

**Comment by nabijaczleweli** [comment]
3-week bump

File: contrib/dracut/90zfs/zfs-lib.sh.in [hunk 2]
```diff
@@ -128,10 +128,8 @@ export_all() {
     ret=0
 
     IFS="${NEWLINE}"
-    for pool in $(zpool list -H -o name) ; do
-        if zpool list -H "${pool}" > /dev/null 2>&1; then
-            zpool export "${pool}" "$@" || ret=$?
-        fi
+    for pool in $(zpool list -Ho name); do
+        zpool export "${pool}" "$@" || ret=$?
     done
```
Comment (behlendorf) [inline contrib/dracut/90zfs/zfs-lib.sh.in:133]: Is there a reason we can't use `zpool export -a` and get rid of the loop entirely?

```
     -a      Exports all pools imported on the system.
```

**Review by behlendorf — APPROVED** [review]

File: contrib/dracut/90zfs/zfs-lib.sh.in (hunk 2 above)
Comment (nabijaczleweli) [reply contrib/dracut/90zfs/zfs-lib.sh.in]: Historically, yes: 07a3312f170ac56cb480b0df9fdf4c83f116b59b, which originated this, landed on Tue Oct 7 10:20:56 2014 -0700, at which point `zpool export` didn't have `-a`.

Now, though, no. I've replaced the two users with straight `zpool export -aF`.

**Pull request closed without merging**